apps/ensrainbow/src/index.ts

```diff
@@ -24,20 +27,21 @@ try {
 }
 
 app.get("/v1/heal/:labelhash", async (c: Context) => {
-  const labelhash = c.req.param("labelhash")?.toLowerCase();
+  const labelhash = c.req.param("labelhash");
+  const prefixedLabelHash = (labelhash.startsWith('0x') ? labelhash : `0x${labelhash}`) as `0x${string}`;
 
-  if (!labelhash) {
-    return c.json({ error: "Missing labelhash parameter" }, 400);
-  }
-
-  if (!/^(0x)?[0-9a-f]{64}$/.test(labelhash)) {
-    return c.json({ error: "Invalid labelhash - must be a 32 byte hex string" }, 400);
+  let labelHashBytes: Uint8Array;
+  try {
+    labelHashBytes = hexToBytes(prefixedLabelHash);
+    if (labelHashBytes.length !== 32) {
+      return c.json({ error: "Invalid labelhash - must be a 32 byte hex string" }, 400);
+    }
+  } catch (error) {
+    return c.json({ error: "Invalid labelhash - must be a valid hex string" }, 400);
   }
 
   try {
-    // Convert hex string to Buffer, stripping '0x' prefix if present
-    const hashBytes = Buffer.from(labelhash.replace(/^0x/, ""), "hex");
-    const label = await db.get(hashBytes);
+    const label = await db.get(Buffer.from(labelHashBytes));
```

Comment:
now we can do `const label = await db.get(Buffer.from(hexToBytes(labelhash)))` without additional checks